src/cli/helpers.js

```diff
@@ -437,33 +497,126 @@ function getConvertor(value, name) {
 }
 
 /**
- * Converts a set of arguments to {@link rocConfigSettings} object.
+ * Converts a set of options to {@link rocConfigSettings} object and command specific options.
  *
- * @param {Object} args - Arguments parsed from minimist.
+ * @param {Object} options - Options parsed from minimist.
  * @param {Object} mappings - Result from {@link getMappings}.
+ * @param {Object} command - A command.
  *
  * @returns {Object} - The mapped Roc configuration settings object.
  */
-export function parseArguments(args, mappings) {
-    const config = {};
-    const info = [];
+export function parseOptions(options, mappings, command) {
+    const configuration = {};
+    const infoSettings = [];
+    const infoOptions = [];
+
+    let notManagedOptions = {};
+    let possibleCommandOptions = [];
+    let possibleCommandOptionsShort = [];
+    const parsedOptions = {
+        options: {},
+        rest: {}
+    };
+
+    const getName = (name) => name.length === 1 ? '-' + name : '--' + name;
 
-    Object.keys(args).forEach((key) => {
+    // Always match settings options first
+    Object.keys(options).forEach((key) => {
         if (mappings[key]) {
-            const value = convert(args[key], mappings[key]);
-            set(config, mappings[key].path, value);
+            const value = convert(options[key], mappings[key]);
+            set(configuration, mappings[key].path, value);
         } else {
             // We did not find a match
-            info.push(getSuggestions([key], Object.keys(mappings), true));
+            notManagedOptions = {
+                ...notManagedOptions,
+                [key]: options[key]
+            };
         }
     });
 
-    if (info.length > 0) {
-        console.log(errorLabel('CLI problem'), 'Some commands were not understood.\n');
-        console.log(info.join('\n') + '\n');
+    if (command && command.options) {
```

Comment:
This is a pretty large nested control structure. Would prefer if it could be split up a bit more.